src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -1405,76 +954,12 @@ void IrcMessageHandler::handlePartMessage(Communi::IrcMessage *message)
     }
 }
 
-float IrcMessageHandler::similarity(
-    const MessagePtr &msg, const LimitedQueueSnapshot<MessagePtr> &messages)
-{
-    float similarityPercent = 0.0F;
-    int checked = 0;
-
-    for (int i = 1; i <= messages.size(); ++i)
-    {
-        if (checked >= getSettings()->hideSimilarMaxMessagesToCheck)
-        {
-            break;
-        }
-        const auto &prevMsg = messages[messages.size() - i];
-        if (prevMsg->parseTime.secsTo(QTime::currentTime()) >=
-            getSettings()->hideSimilarMaxDelay)
-        {
-            break;
-        }
-        if (getSettings()->hideSimilarBySameUser &&
-            msg->loginName != prevMsg->loginName)
-        {
-            continue;
-        }
-        ++checked;
-        similarityPercent = std::max(
-            similarityPercent,
-            relativeSimilarity(msg->messageText, prevMsg->messageText));
-    }
-
-    return similarityPercent;
-}
-
-void IrcMessageHandler::setSimilarityFlags(const MessagePtr &message,
-                                           const ChannelPtr &channel)
-{
-    if (getSettings()->similarityEnabled)
-    {
-        bool isMyself =
-            message->loginName ==
-            getApp()->getAccounts()->twitch.getCurrent()->getUserName();
-        bool hideMyself = getSettings()->hideSimilarMyself;
-
-        if (isMyself && !hideMyself)
-        {
-            return;
-        }
-
-        if (IrcMessageHandler::similarity(message,
-                                          channel->getMessageSnapshot()) >
-            getSettings()->similarityPercentage)
-        {
-            message->flags.set(MessageFlag::Similar, true);
-            if (getSettings()->colorSimilarDisabled)
-            {
-                message->flags.set(MessageFlag::Disabled, true);
-            }
-        }
-    }
-}
-
 void IrcMessageHandler::addMessage(Communi::IrcMessage *message,
-                                   const ChannelPtr &chan,
-                                   const QString &originalContent,
-                                   ITwitchIrcServer &server, bool isSub,
```

Comment:
Undo the changes to `ITwitchIrcServer` and `ChannelPtr` here please, unless they are necessary for the deduplication

Reply:
Re-added the server, but there's no point in using a `ChannelPtr` there. We're dealing with `TwitchChannel`s and the signatures should show that.

Reply:
Since this is not a tiny refactor, I wanted as few things changed as possible unless they were directly related to the IRC/RecentMessages parsing. Function signature changes & renames like `chan` to `channel` makes this more difficult.
If changing `ChannelPtr` to `TwitchChannel` makes sense as part of this refactor (e.g. because RecentMessages paths already deal directly with a `TwitchChannel`), at least keep the variable name the same.